### PR TITLE
Fixes #2243 - Crash on item selection

### LINKED ISSUE #2243: Crash on item selection
## Exception
* User Action: UI Error
* Request: Application crash
* Version: 3.5.3
* OS: Linux Android 8.1.0 - 27
* Device: TB-8704X
* Model: Lenovo TB-8704X
* Product: TB-8704X
<details><summary><b>Crash log </b></summary><p>

java.lang.ArrayIndexOutOfBoundsException: length=22; index=-1
 at java.util.ArrayList.get(ArrayList.java:439)
 at com.amaze.filemanager.adapters.RecyclerAdapter.toggleChecked(RecyclerAdapter.java:179)
 at com.amaze.filemanager.ui.fragments.MainFragment.onListItemClicked(MainFragment.java:799)
 at com.amaze.filemanager.adapters.RecyclerAdapter.lambda$onBindViewHolder$0(RecyclerAdapter.java:552)
 at com.amaze.filemanager.adapters.RecyclerAdapter.lambda$onBindViewHolder$0$RecyclerAdapter(Unknown Source:0)
 at com.amaze.filemanager.adapters.-$$Lambda$RecyclerAdapter$FDaxlO2LtK_Q0Z-YEozjkaXX_T8.onClick(Unknown Source:11)
 at android.view.View.performClick(View.java:6294)
 at android.view.View$PerformClick.run(View.java:24774)
 at android.os.Handler.handleCallback(Handler.java:790)
 at android.os.Handler.dispatchMessage(Handler.java:99)
 at android.os.Looper.loop(Looper.java:164)
 at android.app.ActivityThread.main(ActivityThread.java:6518)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

</details>
<hr>

Raised by: shuvashish76
Reporter: shuvashish76

### PR DESCRIPTION
**Pull request by VishnuSanal**
#### Issue tracker   
Fixes will automatically close the related issue

Fixes #2243

#### Release  
Addresses release/3.6
  
#### Test cases
- [ ] Covered
  
#### Manual testing
- [ ] Done

#### Build tasks success  
Successfully running following tasks on local 
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

File: app/src/main/java/com/amaze/filemanager/adapters/RecyclerAdapter.java [hunk 1]
```diff
@@ -176,6 +177,11 @@ public RecyclerAdapter(
    * @param imageView the check {@link CircleGradientDrawable} that is to be animated
    */
   public void toggleChecked(int position, ImageView imageView) {
+    if (itemsDigested.size() <= position) {
```
Comment (EmmanuelMess) [inline app/src/main/java/com/amaze/filemanager/adapters/RecyclerAdapter.java:180]: Why is the position larger than the item size? are there too few items or does something calculate the item clicked incorrectly?

Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/adapters/RecyclerAdapter.java:180]: Item position is directly being fetched from `viewholder.getAdapterPosition` in `onBindViewHolder`
Not sure why it's returning position greater than the recyclerview adapter size.

**Review by VishalNehra — DISMISSED** [review]

File: app/src/main/java/com/amaze/filemanager/adapters/RecyclerAdapter.java (hunk 1 above)
Comment (VishnuSanal) [reply app/src/main/java/com/amaze/filemanager/adapters/RecyclerAdapter.java:180]: But @VishalNehra, I have a concern, as @EmmanuelMess suggests, the error shows `java.lang.ArrayIndexOutOfBoundsException: length=22; index=-1` means the requested index is `-1` and not something greater than the length of the array!

Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/adapters/RecyclerAdapter.java:180]: Very well, can put a check for index < 0 as well.

**Review by VishalNehra — APPROVED** [review]

**Pull request merged**